### PR TITLE
Deliverable 3

### PR DESCRIPTION
**Pull request by AlanMalagon**
-The search term is stored and retrieved from the Global Context
-The appearance theme is stored in the Global Context and now working.
-User reducer handles the value of both search term and theme.
-Test cases not covered before were implemented and fixed to fit the new implementations

### CODE REVIEW

File: src/components/Header/DarkmodeButton/DarkmodeButton.test.js [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
+        </GlobalProvider>)
         const button = container.querySelector('button');
         expect(button).toBeInTheDocument();
     });
     
     it("the button should include an svg", ()=>{
-        render(<DarkmodeButton/>);
+        render(<GlobalProvider>
+            <DarkmodeButton/>
+        </GlobalProvider>);
```
Comment (PacoMojica) [inline src/components/Header/DarkmodeButton/DarkmodeButton.test.js:19]: Great I like to see that you are updating the tests with the new stuff

File: src/reducer.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    switch(action.type){
+        case 'theme':
+            theme.name === 'dark' ? theme = light : theme = dark;
+            storage.set('theme', JSON.stringify(theme));
+            return {...state, theme };
+        case 'search':
+            return {...state, search: action.value}
+        default:
+            return new Error(`Unhandled action ${action.type} in reducer was found`);
```
Comment (PacoMojica) [inline src/reducer.js:15]: A simple but effective reducer 👍

File: test-coverage.txt [hunk 3]
```diff
@@ -1,43 +1,78 @@
...
-  VideosGrid.component.jsx           |      100 |      100 |      100 |      100 |                   |
- components/VideosGrid/VideoGridItem |      100 |      100 |      100 |      100 |                   |
-  VideoGridItem.component.jsx        |      100 |      100 |      100 |      100 |                   |
- pages/Home                          |     62.5 |       50 |       50 |     62.5 |                   |
-  Home.page.jsx                      |     62.5 |       50 |       50 |     62.5 |          17,18,19 |
--------------------------------------|----------|----------|----------|----------|-------------------|
+------------------------------------------------------------|----------|----------|----------|----------|-------------------|
+File                                                        |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
+------------------------------------------------------------|----------|----------|----------|----------|-------------------|
+All files                                                   |    78.83 |    58.33 |    76.39 |    79.19 |                   |
```
Comment (PacoMojica) [inline test-coverage.txt:4]: The test coverage is looking good

**Review by PacoMojica — COMMENTED** [review]
Hello Alan, your challenge has all the requirements for this delivery including the bonus, great work!

## Acceptance Criteria
- [x] The search term is stored and retrieved from the Global Context correctly.
- [x] The appearance theme is stored on the Global Context and applied correctly to the App UI.
- [x] `useReducer` hook is implemented correctly to manage the Global State.

## Bonus Points
- [x] Testing coverage is above 70%. (Please include a screenshot of the code coverage report in your PR description).

**Comment by AlanMalagon** [comment]
Thank you for the feedback, Paco. Have a great day.

**Pull request merged**